src/c-dvar-reader.c

```diff
@@ -436,6 +436,21 @@ static int c_dvar_ff(CDVar *var) {
         do {
                 if (var->current->n_type && (var->current->container != 'a' || c_dvar_more(var))) {
                         c = var->current->i_type->element;
+                        if (var->current->container == 'a'
+                                && var->current->i_type->size > 0
+                                && var->current->i_type->basic
+                                && c != 'b') {
+                                /* This is an array of fixed size basic elements other than booleans.
+                                 * We can skip it right now.
+                                 * We just check if all the elements fit in the buffer.
+                                 */
+                                if ((var->current->n_buffer % var->current->i_type->size) != 0)
+                                        return C_DVAR_E_CORRUPT_DATA;
+
+                                var->current->i_buffer += var->current->n_buffer;
+                                var->current->n_buffer = 0;
+                                c = ']';
+                        }
                 } else {
                         switch (var->current->container) {
                         case 'a':
```

Comment:
Rather than supporting all fixed-size, builtin types, and then using a blacklist, how about just whitelisting anything that works? This is more explicit and less error prone, I think. So how about this:

```
@@ -480,6 +484,32 @@ static int c_dvar_ff(CDVar *var) {
                         assert(depth > 0);
                         --depth;
                         break;
+                case 'y':
+                case 'n':
+                case 'q':
+                case 'i':
+                case 'h':
+                case 'u':
+                case 'x':
+                case 't':
+                case 'd':
+                        /*
+                         * If we are skipping an entire array with a fixed size
+                         * member, we can jump over all members in one go. Note
+                         * that this only works if the member does not need any
+                         * validation (this excludes 'b' and any non-basic
+                         * type), because callers rely on this function to
+                         * check for content validity.
+                         */
+                        if (depth > 0 && var->current->container == 'a') {
+                                t = var->current->n_buffer % var->current->i_type->size;
+                                var->current->i_buffer += var->current->n_buffer - t;
+                                var->current->n_buffer = t;
+
+                                c = ']';
+                                --depth;
+                        }
+                        break;
                 }
 
                 r = c_dvar_read(var, (char [2]){ c, 0 }, NULL);
```

Reply:
Looks good, with just minor fix 
+                                t = var->current->n_buffer % var->current->i_type->size;
should be
+                                int t = var->current->n_buffer % var->current->i_type->size;

I'm kinda new to github, so... what now? Should I take it to my branch for updating the pull request?

Reply:
I think every project does this differently. I usually just change the commits (rebase/rewrite) and then force-push the branch. This will automatically update the PR.

However, I am also fine with amending the patches myself, or pushing follow-ups (especially if it is just a small change).

Reply:
I've copied the above code, added 'int', set @dvdhrm as the author, and force-pushed.